src/libstd/io/tempfile.rs

```diff
@@ -78,47 +80,59 @@ pub struct TempDir {
     disarmed: bool
 }
 
+// How many times should we (re)try finding an unused random name? It should be
+// enough that an attacker will run out of luck before we run out of patience.
+const NUM_RETRIES: uint = 1 << 31;
```

Comment:
This seems like quite a large constant! If making a directory took one millisecond then this would cause the function to take a maximum of 25 days! I do highly doubt that we'll ever hit that, but do you have a link to the source with this large a number of retries? I'm curious to just poke around it a bit.